test/distributed/org/apache/cassandra/distributed/action/GossipHelper.java

```diff
@@ -0,0 +1,431 @@
+    }
+
+    public static VersionedApplicationState statusBootstrapping(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.STATUS, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).bootstrapping(tokens));
+    }
+
+    public static VersionedApplicationState statusWithPortBootstrapping(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.STATUS_WITH_PORT, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).bootstrapping(tokens));
+    }
+
+    public static VersionedApplicationState statusLeaving(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.STATUS, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).leaving(tokens));
+    }
+
+    public static VersionedApplicationState statusWithPortLeaving(IInvokableInstance instance)
+    {
+        return versionedToken(instance, ApplicationState.STATUS_WITH_PORT, (partitioner, tokens) -> new VersionedValue.VersionedValueFactory(partitioner).leaving(tokens));
+    }
+
+    public static VersionedValue toVersionedValue(VersionedApplicationState vv)
+    {
+        return VersionedValue.unsafeMakeVersionedValue(vv.value, vv.version);
+    }
+
+    public static ApplicationState toApplicationState(VersionedApplicationState vv)
+    {
+        return ApplicationState.values()[vv.applicationState];
+    }
+
+    public static VersionedApplicationState unsafeVersionedToken(IInvokableInstance instance,
+                                                                 ApplicationState applicationState,
+                                                                 SerializableBiFunction<IPartitioner, Collection<Token>, VersionedValue> supplier,
+                                                                 String partitionerStr, String initialTokenStr)
+    {
+        return instance.appliesOnInstance((String partitionerString, String tokenString) -> {
+            IPartitioner partitioner = FBUtilities.newPartitioner(partitionerString);
+            Token token = partitioner.getTokenFactory().fromString(tokenString);
+
+            VersionedValue versionedValue = supplier.apply(partitioner, Collections.singleton(token));
+            return new VersionedApplicationState(applicationState.ordinal(), versionedValue.value, versionedValue.version);
+        }).apply(partitionerStr, initialTokenStr);
+    }
+
+    public static VersionedApplicationState versionedToken(IInvokableInstance instance, ApplicationState applicationState, SerializableBiFunction<IPartitioner, Collection<Token>, VersionedValue> supplier)
+    {
+        return unsafeVersionedToken(instance, applicationState, supplier, instance.config().getString("partitioner"), instance.config().getString("initial_token"));
+    }
+
+    public void removeFromRing(IInstance peer)
```

Comment:
Thanks for spotting. You're right, this did throw because it wasn't wrapped into `InstanceAction`: I was initially using this for testing forced removal from the ring. I've added `nodeNotInRingTest` for this.